src/basic/architecture.c

```diff
@@ -118,20 +118,19 @@ int uname_architecture(void) {
                 { "arc",        ARCHITECTURE_ARC      },
                 { "arceb",      ARCHITECTURE_ARC_BE   },
 #else
-#error "Please register your architecture here!"
+#  error "Please register your architecture here!"
 #endif
         };
 
         static int cached = _ARCHITECTURE_INVALID;
         struct utsname u;
-        unsigned i;
 
         if (cached != _ARCHITECTURE_INVALID)
                 return cached;
 
         assert_se(uname(&u) >= 0);
 
-        for (i = 0; i < ELEMENTSOF(arch_map); i++)
+        for (unsigned i = 0; i < ELEMENTSOF(arch_map); i++)
```

Comment:
size_t?

Reply:
Considering that this is a hand-written map, I think it's unlikely we'll go above 4e9 entries ;) But I'll change it.